src/main/java/com/upserve/uppend/BlockedLongs.java

```diff
@@ -420,32 +424,17 @@ public void clear() {
     public void close() throws IOException {
         log.debug("closing {}", file);
 
-        if (readOnly) {
-            blocks.close();
-            return;
-        }
+        Arrays.fill(pages, null);
 
-        IntStream.range(0, LOCK_SIZE).forEach(index -> stripedLocks.getAt(index).lock());
-        try {
-            flush();
-            blocks.close();
-        } finally {
-            IntStream.range(0, LOCK_SIZE).forEach(index -> stripedLocks.getAt(index).unlock());
-        }
+        flush();
+        blocks.close();
     }
 
     @Override
     public void flush() {
         if (readOnly) return;
         log.debug("flushing {}", file);
-        posBuf.force();
         appendCountBuf.putLong(0, initialAppendCount + appendCounter.sum());
-        appendCountBuf.force();
-
-        Arrays.stream(pages)
-                .parallel()
-                .filter(Objects::nonNull)
-                .forEach(MappedByteBuffer::force);
```

Comment:
Why are you no longer flushing pages inside this method called `flush`?

Reply:
Because flush guarantees durability - written to disk, but I now understand that is not required for multiple processes to share state in the same machine through the page cache.